cmd/main.go

```diff
@@ -190,9 +194,29 @@ func setupProxmoxClient(ctx context.Context, logger logr.Logger) (capmox.Client,
 	if ProxmoxURL == "" || ProxmoxTokenID == "" || ProxmoxSecret == "" {
 		return nil, nil
 	}
-	// TODO, check if we need to delete tls config
+
+	rootCerts, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, fmt.Errorf("loading system cert pool: %w", err)
+	}
+
+	if proxmoxRootCertFile != "" {
+		// There is a custom root cert, we need to load it
+		pemBlock, err := os.ReadFile(proxmoxRootCertFile) //#nosec:G304 // File provided by system operator
+		if err != nil {
+			return nil, fmt.Errorf("reading proxmox-root-cert-file: %w", err)
+		}
+
+		if !rootCerts.AppendCertsFromPEM(pemBlock) {
```

Comment:
Not sure this is supposed to load a chain… …normally you'd only load one cert into a root pool (or multiple non-chained in case you're using different root CAs)… Need to look this up.

Reply:
Hm I suppose you **could** import a chain but that makes no sense. Importing the top-most cert of the chain is already sufficient to allow-list the whole chain. I'd keep this as non-chain but single-cert.